### PR TITLE
Loading and Saving snippets

### PR DESCRIPTION
**Pull request by toledoal**
Hello! In this PR I'm adding the loading and saving snippets capabilities in the Curve Editor

For this functionality I'm currently missing:

1. Notify observers across application
2. Update state in curve editor to reflect recently uploaded animations.

Also, I need to test if I can save a snippet to server, but I'm having trouble with the CORS policy in local dev. So the idea is to test once this is deployed. This will also help to test loading snippets since the playground adds a 'code' field when trying to save JSON as snippets.

### CODE REVIEW

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx [hunk 1]
```diff
@@ -1410,6 +1413,8 @@ export class AnimationCurveEditorComponent extends React.Component<
               setNotificationMessage={(message: string) => {
                 this.setState({ notification: message });
               }}
+              globalState={this.props.globalState}
+              snippetServer={this._snippetUrl}
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationCurveEditorComponent.tsx:1417]: can we call it snippetId?

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx [hunk 2]
```diff
@@ -1,49 +1,124 @@
...
   }
 
   change(value: string) {
-    this.setState({ server: value });
+    this.setState({ snippetId: value });
+    this.props.setSnippetId(value);
+  }
+
+  loadFromFile(file: File) {
+    Tools.ReadFile(
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx:43]: you should use Animation.ParseFromFileAsync

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx [hunk 3]
```diff
@@ -1,49 +1,124 @@
...
+          label='Snippet Id'
           lockObject={this.props.lockObject}
-          value={this.state.server}
+          value={this.state.snippetId}
           onChange={(value: string) => this.change(value)}
         />
-        <ButtonLineComponent label="Load" onClick={() => {}} />
-        <div className="load-browse">
+        <ButtonLineComponent
+          label='Load'
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx:108]: please reuse the same naming as in the particle editor (I guess it was "Load from Snippet Server"

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/saveSnippet.tsx [hunk 4]
```diff
@@ -51,38 +60,124 @@ export class SaveSnippet extends React.Component<
...
             })}
           </ul>
         </div>
-        <div className="save-buttons">
-          <ButtonLineComponent label="Save Snippet" onClick={() => {}} />
-          <ButtonLineComponent label="Save File" onClick={() => {}} />
+        <div className='save-buttons'>
+          {this.props.snippetId !== '' ? (
+            <ButtonLineComponent
+              label='Save Snippet'
```
Comment (deltakosh) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/saveSnippet.tsx:169]: same here. let's make it consistent with particle editor

**Comment by toledoal** [comment]
I'll check the linting issue as well

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx (hunk 2 above)
Comment (toledoal) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx:43]: Hi!, A question about this... I was using Tools for local files. ParseFromFileAsync asks me for a url, should I create another option to load from a remote location? 
image

Comment (deltakosh) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx:43]: ignore me :) I mixed stuff:) Your code is fine !

Comment (toledoal) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx:43]: Maybe there is a third option to load animations from. 😄

Comment (deltakosh) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/loadsnippet.tsx:43]: totally but we do not expose it in the UI so all good :D

**Pull request merged**